### PR TITLE
PT-2184: Simplify contact owner form

### PR DESCRIPTION
**Pull request by mjshepherd**
@buske @allasm please review

### CODE REVIEW

File: communication/src/main/java/org/phenotips/messaging/internal/DefaultActionManager.java [hunk 1]
```diff
@@ -70,6 +71,8 @@
 
     private static final String SUBJECT = "Access to patient record granted";
 
+    private static final String SUBJECT_STRING = "subject";
```
Comment (buske) [inline communication/src/main/java/org/phenotips/messaging/internal/DefaultActionManager.java]: I would suggest `_KEY` or `_FIELD`.

**Comment by buske** [comment]
Is the initiating user included as the reply-to address (or are there contact details guaranteed to be included somewhere in the email)?

**Comment by buske** [comment]
There's a missing space in the template here: `My patient is undiagnosedand presents the following phenotypic features`

**Comment by buske** [comment]
The `To:` form is also left blank:
screenshot

**Comment by buske** [comment]
The form does not properly handle errors in email sending. Even when sending the email failed (e.g. SMTP server is not set up), it still redirects to a modal with: `Your email has been successfully sent.`

**Comment by buske** [comment]
The subject of the email is also missing a space: `[PhenomeCentral]Interested in one of your non-public cases`

**Comment by buske** [comment]
Interesting, because the email is sent as both plaintext and HTML, but with an empty HTML part, the email displays as **empty** on mail clients that are set to view as HTML. The content is still there in the text version, but isn't displayed:
```
Content-Type: multipart/mixed; 
	boundary="----=_Part_14_1816969756.1454562429382"

------=_Part_14_1816969756.1454562429382
Content-Type: multipart/alternative; 
	boundary="----=_Part_16_1000452234.1454562429382"

------=_Part_16_1000452234.1454562429382
Content-Type: text/plain; charset=us-ascii
Content-Transfer-Encoding: 7bit

Hello,

PhenomeCentral has identified significant similarities between one of your cases and one of mine. I would like to discuss our cases.

My patient is diagnosed with
* #143100 HUNTINGTON DISEASE; HD
* #215700 CITRULLINEMIA, CLASSIC
and presents the following phenotypic features:
* Camptodactyly of finger
* Coarctation of aorta
* Seizures
* Spasticity

Please let me know if you are interested in my case.


------=_Part_16_1000452234.1454562429382
Content-Type: multipart/related; 
	boundary="----=_Part_15_762404106.1454562429382"

------=_Part_15_762404106.1454562429382
Content-Type: text/html; charset=UTF-8
Content-Transfer-Encoding: quoted-printable
Content-Disposition: inline


------=_Part_15_762404106.1454562429382--

------=_Part_16_1000452234.1454562429382--

------=_Part_14_1816969756.1454562429382--
```

File: similarity-data-impl/src/main/java/org/phenotips/data/similarity/internal/AbstractVariant.java [hunk 2]
```diff
@@ -21,6 +21,7 @@
 
 import java.util.Map;
 
+import org.json.JSONException;
```
Comment (buske) [inline similarity-data-impl/src/main/java/org/phenotips/data/similarity/internal/AbstractVariant.java]: Unnecessary import

**Comment by buske** [comment]
```
[INFO] There are 3 errors reported by Checkstyle 6.11.2 with checkstyle.xml ruleset.
[ERROR] src/main/java/org/phenotips/data/similarity/internal/AbstractPatientSimilarityView.java:[45,48] (coding) MultipleStringLiterals: The String "id" appears 2 times in the file.
[ERROR] src/main/java/org/phenotips/data/similarity/internal/AbstractPatientSimilarityView.java:[46,50] (coding) MultipleStringLiterals: The String "name" appears 2 times in the file.
[ERROR] src/main/java/org/phenotips/data/similarity/internal/AbstractVariant.java:[24,8] (imports) UnusedImports: Unused import - org.json.JSONException.
```

**Comment by buske** [comment]
To: `PhenomeCentral at $affiliation`

**Comment by buske** [comment]
If user does not have a name, should fall back on username instead of an empty To box.

File: communication/src/main/java/org/phenotips/messaging/script/AnonymousCommunicationScriptService.java [hunk 3]
```diff
@@ -55,17 +55,13 @@
      * @param connectionId the id of the anonymous communication linking the two patients and their owners that are
      *            involved in this connection
      * @param options the mail content options selected by the user
-     * @return {@code 0} if the mail was successfully sent, other numbers in case of errors
+     * @return {@code 0} if the mail was successfully sent, http response codes in case of errors
      */
     public int sendInitialMail(String connectionId, Map<String, Object> options)
     {
-        try {
```
Comment (sdumitriu) [inline communication/src/main/java/org/phenotips/messaging/script/AnonymousCommunicationScriptService.java]: Why was the try catch removed?

Comment (mjshepherd) [inline communication/src/main/java/org/phenotips/messaging/script/AnonymousCommunicationScriptService.java]: It didn't seem to be catching anything really. I added some more specific catch statements inside of the ``actionManager.sendInitialMails`` call. That way it returns some http codes on failure

**Pull request merged**